### PR TITLE
Refactor config loading

### PR DESCRIPTION
**Pull request by jdavcs**
This addresses #8493 

SUMMARY
1. `config/__init__.py`:
- Config properties are loaded from `config_schema.yml`
- Default values are overwritten by values from keyword args. Values are cast as int/float/bool based on types specified in the schema; str is the default
- Attributes are created for all properties defined in the schema
- If `foo` is defined in the schema, then statements like `foo = kwargs.get('foo', [default])` have been mostly deleted (see 2e61a7a commit message details)
- Comments for the above statements have been deleted if redundant; otherwise, added to the property's description in the schema

2. `config_schema.yml`:
- Removed empty string default values if those defaults were not assigned (`None` was assigned instead) 
- Minor edits; changed type int to float to be consistent with the `float()` cast in `config/`

3. Relevant unit tests added.

NOTES
1. Attributes are created for all properties defined in the schema; not all of them are necessary. However, this automated approach (a) eliminates the need for numerous individual assignment statements; (b) fixes inconsistencies between schema and `config/`; and (c) is a step towards making `__init__()` more readable and testable. 

2. There are more details on each commit in the commit messages.

3. This is a work in progress. I'd like to get this merged first, if possible, since this is the foundation of all subsequent edits.

(ping: @jmchilton @mvdbeek @natefoo , as we've discussed parts of this)

### CODE REVIEW

**Comment by martenson** [comment]
@galaxybot test this

File: lib/galaxy/config/__init__.py [hunk 1]
```diff
@@ -679,9 +574,12 @@ def __init__(self, **kwargs):
...
         self.default_locale = kwargs.get('default_locale', None)
-        self.master_api_key = kwargs.get('master_api_key', None)
-        if self.master_api_key == "changethis":  # default in sample config file
-            raise ConfigurationError("Insecure configuration, please change master_api_key to something other than default (changethis)")
+
+        # TODO: the following statement is commented out temporarily. Previously, the 'changethis' default value was never used
+        #   (instead, None was assigned by default). Now that defaults specified in the schema are used, this raises an error during testing.
+        #   One solution is to mock this for testing.
+        # if self.master_api_key == "changethis":  # default in sample config file
+        #     raise ConfigurationError("Insecure configuration, please change master_api_key to something other than default (changethis)")
```
Comment (jmchilton) [inline lib/galaxy/config/__init__.py]: Change the default in the schema to ``None`` and drop the his whole section?

Comment (jdavcs) [reply lib/galaxy/config/__init__.py]: Done

**Comment by jdavcs** [comment]
@jmchilton should I rebase/resolve conflict?

**Comment by dannon** [comment]
@ic4f In general, yes.  This one's pretty minimal, just looks to be the changes from #8574

**Comment by jdavcs** [comment]
Thanks, will do.

**Comment by jdavcs** [comment]
@galaxybot test this

**Comment by jdavcs** [comment]
Thank you!

**Pull request merged**